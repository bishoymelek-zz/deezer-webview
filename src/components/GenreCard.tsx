import React from 'react';
import { Card } from 'antd';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

type GenreCardPropType = {
  name: string;
  id: string;
  picture: string;
  className?: string;
};

function GenreCard(props: GenreCardPropType): JSX.Element {
  const { picture, name, id, className } = props;
  return (
    <Link to={`/genres/${id}`}>
      <Card
        className={`genre-card ${className}`}
        hoverable
        title={name}
        cover={<img alt="genre" src={picture} />}
      />
    </Link>
  );
}

const StyledGenreCard = styled(GenreCard)`
  &.genre-card {
    .ant-card-head-title {
      text-align: center;
    }
  }
`;

export default StyledGenreCard;
