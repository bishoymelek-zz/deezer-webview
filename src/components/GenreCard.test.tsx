import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import GenreCard from './GenreCard';

const mockStore = configureStore([]);

describe('Component/GenreCard', () => {
  let store;
  let component;
  const genreCardMockData = {
    name: 'Hi',
    id: '1',
    picture: 'pic.png',
  };
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <GenreCard
            name={genreCardMockData.name}
            id={genreCardMockData.id}
            picture={genreCardMockData.picture}
          />
        </BrowserRouter>
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
