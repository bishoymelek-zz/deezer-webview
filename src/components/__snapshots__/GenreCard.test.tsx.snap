// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/GenreCard should render with given state from Redux store 1`] = `
<a
  href="/genres/1"
  onClick={[Function]}
>
  <div
    className="ant-card genre-card ant-card-bordered ant-card-hoverable"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Hi
        </div>
      </div>
    </div>
    <div
      className="ant-card-cover"
    >
      <img
        alt="example"
        src="pic.png"
      />
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    />
  </div>
</a>
`;
